freetextresponse/freetextresponse.py

```diff
@@ -464,40 +503,64 @@ def _get_submitted_message(self):
         Returns the message to display in the submission-received div
         """
         result = ''
-        if self.count_attempts > 0 and self._word_count_valid():
+        if self._word_count_valid():
             result = self.submitted_message
         return result
 
+    def _get_user_alert(self, ignore_attempts=False):
+        """
+        Returns the message to display in the user_alert(TBD) div
+        """
+        result = ''
+        if not self._word_count_valid():
+            result = self._get_word_count_message(ignore_attempts)
+        return result
+
     @XBlock.json_handler
     def submit(self, data, suffix=''):
         # pylint: disable=unused-argument
         """
         Processes the user's submission
         """
-        if self.max_attempts > 0 and self.count_attempts >= self.max_attempts:
-            raise StandardError(
-                _(
-                    'User has already exceeded the '
-                    'maximum number of allowed attempts'
-                )
-            )
-        self.student_answer = data['student_answer']
-        if self._word_count_valid():
-            if self.max_attempts == 0:
-                self.count_attempts = 1
-            else:
-                self.count_attempts += 1
-            self._compute_score()
+        if self.max_attempts == 0 or self.count_attempts < self.max_attempts:
+            self.student_answer = data['student_answer']
+            if self._word_count_valid():
+                if self.max_attempts == 0:
+                    self.count_attempts = 1
```

Comment:
I was just pointing out that I have no idea why the code was written this way because I didn't write it.   I'm not worried about integer overflow, I just think that could have been a reason.    

I'm not altering production code with out a good reason.  If you can explain why pinning count_attempts is wrong then I will change it.